static/app/components/modals/recoveryOptionsModal.tsx

```diff
@@ -1,118 +1,116 @@
-import {Fragment} from 'react';
+import {Fragment, useState} from 'react';
 
 import type {ModalRenderProps} from 'sentry/actionCreators/modal';
 import {Alert} from 'sentry/components/alert';
 import {Button} from 'sentry/components/button';
-import DeprecatedAsyncComponent from 'sentry/components/deprecatedAsyncComponent';
 import {t} from 'sentry/locale';
 import {space} from 'sentry/styles/space';
 import type {Authenticator} from 'sentry/types';
+import {useApiQuery} from 'sentry/utils/queryClient';
 import TextBlock from 'sentry/views/settings/components/text/textBlock';
 
-type Props = DeprecatedAsyncComponent['props'] &
-  ModalRenderProps & {
-    authenticatorName: string;
-  };
-
-type State = DeprecatedAsyncComponent['state'] & {
-  authenticators: Authenticator[] | null;
-  skipSms: boolean;
+type Props = ModalRenderProps & {
+  authenticatorName: string;
 };
 
-class RecoveryOptionsModal extends DeprecatedAsyncComponent<Props, State> {
-  getDefaultState() {
-    return {...super.getDefaultState(), skipSms: false};
-  }
+function RecoveryOptionsModal({
+  authenticatorName,
+  closeModal,
+  Body,
+  Header,
+  Footer,
+}: Props) {
+  const {isLoading, data: authenticators = []} = useApiQuery<Authenticator[]>(
```

Comment:
you'll want to do something with isError as well since DeprecatedAsyncComponent was handling that as well